### DISCUSSION TITLE
v2.0.6

### DISCUSSION THREAD

**Discussion by alexdelprete** [alexdelprete/ha-abb-powerone-pvi-sunspec#27] [Announcements]
- Introduced new model detection, processing the Options register as per FIMER specifications:

![image](https://user-images.githubusercontent.com/7027842/194595381-1104cc34-8b14-4ff9-9a32-430688e190a8.png)

- Restructured the ModBus read routine: split it in 3 reads because on some old inverters there was a problem reading more than 50 registers at a time. Max register per read in ModBus TCP is 123. So now the integration does 3 reads. Might change this in the future, it's not the most efficient way.
- Added some debug logging in case we need it to troubleshoot
- Added error logging on inverter type and model checks, if something strange we expose it in the log as error

<hr /><em>This discussion was created from the release <a href='https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/releases/tag/v2.0.6'>v2.0.6</a>.</em>

**Comment by dexmorgan8888** [comment]
2.0.7 broke mine

> Logger: homeassistant.components.sensor
> Source: custom_components/abb_powerone_pvi_sunspec/__init__.py:279
> Integration: Sensor ([documentation](https://www.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22))
> First occurred: 12:25:20 (3 occurrences)
> Last logged: 15:47:45
> 
> Error while setting up abb_powerone_pvi_sunspec platform for sensor
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
>     await asyncio.shield(task)
>   File "/config/custom_components/abb_powerone_pvi_sunspec/sensor.py", line 21, in async_setup_entry
>     hub.read_sunspec_modbus_data()
>   File "/config/custom_components/abb_powerone_pvi_sunspec/__init__.py", line 193, in read_sunspec_modbus_data
>     return self.read_sunspec_modbus_model_1() and self.read_sunspec_modbus_model_101_103() and self.read_sunspec_modbus_model_160()
>   File "/config/custom_components/abb_powerone_pvi_sunspec/__init__.py", line 279, in read_sunspec_modbus_model_1
>     _LOGGER.error("(read_inv) Model unknown, report to @alexdelprete on the forum the following data: Manuf.: %s - Model: %s - Options: %s - OptModel: %s", self.data["comm_manufact"], self.data["comm_model"], self.data["comm_options"], opt_model)
> KeyError: 'comm_model'

Brand: Fimer
Model: PVS-12.5-TL
Grid rate: PVS-12.5-TL_230-400V-50Hz

It was working fine before the upgrade. Please let me know any other info you need

**Reply by dexmorgan8888** [reply]
Reverting back to 2.0.2 solved the problem

**Comment by alexdelprete** [comment]
It doesn't even start? Can you show me a screenshot of the sensors while it's working? I need to check some values of your inverter.

Also: after 2.0.2, did you try 2.0.4?
